lib/jquery.ui/jquery.ui.languagesuggester.js

```diff
@@ -19,32 +46,46 @@ $.widget( 'ui.languagesuggester', $.ui.suggester, {
 			deferred = $.Deferred(),
 			regex = this._escapeRegex( term ),
 			matcher = new RegExp( regex, 'i' ),
+			subCodeMatcher = new RegExp( '^' + regex + '-', 'i' ),
+			prefixMatcher =  new RegExp( '^' + regex, 'i' ),
 			promoters = [
-				new RegExp( '\\(' + regex + '\\)', 'i' ),
-				new RegExp( '\\(' + regex + '-', 'i' ),
-				new RegExp( '^' + regex, 'i' ),
-				new RegExp( '\\(' + regex, 'i' )
+				function( item ) { return item.code.toLowerCase() === term.toLowerCase(); },
+				function( item ) { return subCodeMatcher.test( item.code ); },
+				function( item ) { return prefixMatcher.test( item.label ); },
+				function( item ) { return prefixMatcher.test( item.code ); },
 			];
 
 		deferred.resolve( $.grep( source, function( item ) {
-			return matcher.test( item );
+			return matcher.test( item.code ) || matcher.test( item.label );
 		} ).sort( function( a, b ) {
 			for( var i = 0; i < promoters.length; i++ ) {
-				var promoterA = promoters[i].test( a ),
-					promoterB = promoters[i].test( b );
+				var promoterA = promoters[i]( a ),
+					promoterB = promoters[i]( b );
 
 				if( promoterA !== promoterB ) {
 					return promoterB - promoterA;
 				}
 			}
 
-			return self._localeCompare( a, b );
+			return self._localeCompare( a.label, b.label );
```

Comment:
Unrelated to this patch, but I think we should order by `code`, not by `label`, since this is the default order we are using for sitelinks in the wiki projects.